File: Cullen Numbers/Cullen_number.java
import java.util.Scanner;
import java.lang.Math;

public class Cullen_number {

    static long cullen_number(long n){
        return n * (long) Math.pow(2, n) + 1;
    }

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		long result;
		
		while(true){
			System.out.printf("Enter a number: ");
			result = in.nextLong();
			if(result > -1){
				break;
			}
			else{
				System.out.printf("Natural number expected!\n");
			}
		}
		
        System.out.println(cullen_number(result));
		in.close();
	}
	
	
}

File: Cullen Numbers/Cullen_number.cpp
#include <iostream>
#include <math.h>
using namespace std;
 
long int cullen_number(long int n){
    return n * pow(2, n) + 1;
}

int main(){
	long int number;

	while(true){
		cout << "Enter a number: ";
		cin >> number;

		if(number > -1){
			break;
		}
		else{
			cout << "Natural number expected!\n";
		}
	}

	cout<<cullen_number(number);
}